ClimsoftVer4/ClimsoftVer4/formMetadata.vb

```diff
@@ -1751,4 +1759,35 @@ Err:
         End If
     End Sub
 
+    Private Sub CombEW_SelectedIndexChanged(sender As Object, e As EventArgs) Handles CombEW.SelectedIndexChanged
+        Dim myLon As Double
+        If CombEW.SelectedItem = "W" Then
+
+            myLon = (-1) * (txtDegreesLon.Text + txtMinutesLon.Text / 60 + txtSecondsLon.Text / 3600)
```

Comment:
@mhabimana - this is looking great.

Again, I'll give the kind of comments that I would receive when working in a software engineering team:

One thing to think about is that you are actually repeating the same formula four times in your code:
    `DD = Direction * (Degrees + Minutes/60 + Seconds/3600)`

This can sometimes lead to problems if someone changes one of the formula, intentionally or accidentally, and they do not change the other three (this would make it very hard to see that there was sometimes a problem).

An alternative approach is called "Don't Repeat Yourself" (DRY)... you would define a new function that contained the formula only once and then call it four times:

```
Private Function DMSToDD(Direction As Char, Deg  As Integer, Min As Integer, Sec As Integer) As Double
    ' Convert value in Degrees, Minutes and Seconds (DMS) to Decimal Degrees (DD)
    ' Direction must be N, S, E or W
    Dim multiplier As Integer = 1
    Dim decimalDegrees As Double
    If Direction='S' OrElse Direction = 'W' Then
        multiplier = -1
    End If
    decimalDegrees = multiplier * (Deg + Min/60 + Sec/3600)
    Return Math.Round(decimalDegrees, 4)
End Function
```
A couple more advantages of separating this into its own function:
1. We can write a check in one place that checkes that Deg, Min, etc. are actually integers (and that the users has not typed in text!)
2. We could start writing [Automatic tests](https://en.wikipedia.org/wiki/Test_automation) that check that our code is still working correctly after it gets modified at some point in the future.

To use your new function, you would do something like:
```
txtLongitude.Text = DMSToDD(CombEW.SelectedItem, txtDegreesLon.Text, txtMinutesLon.Text, txtSecondsLon.Text)
```